superbench/benchmarks/model_benchmarks/pytorch_base.py

```diff
@@ -172,6 +172,33 @@ def _create_optimizer(self):
 
         return True
 
+    def _sync_result(self, result):
+        """Function to reduce the result to rank 0.
+
+        Args:
+            result (list): The result data to sync.
+
+        Return:
+            True if reduce result data successfully.
+        """
+        if not super()._sync_result(result):
+            return False
+
+        try:
+            if self._args.distributed_impl == DistributedImpl.DDP:
+                tensor = torch.as_tensor(result)
+                torch.distributed.reduce(tensor, 0, op=torch.distributed.ReduceOp.MAX)
```

Comment:
did you test this? does it need to create the tensor on gpu?

Reply:
Have tested. Since NCCL backend only support collective API on GPU, so we need to transfer the tensor to GPU first.
But for gloo backend, we should use CPU. Maybe I can add a judgement here.

Reply:
Looks there have change not pushed...
Have pushed.